### PR TITLE
Temp

### PR DESCRIPTION
**Pull request by pariahGH**

### CODE REVIEW

File: pages/Page.vue [hunk 1]
```diff
@@ -1,7 +1,158 @@
...
+    funeral_description: "",
+    funeral_location: "",
+    obituary: "",
+    deadline: new Date(),
+    donation_goal: 0,
+    donated_amount:0,
+    donated_percentage: 0,
+    query: ""
+}) 
+const create_checkup_session =async () => {
```
Comment (pariahGH) [inline pages/Page.vue:49]: You will need a variable to hold the donation form data, and that viarabel will be sent to the backend. See ofeks work in the EditPage/[id].vue file as an example

File: pages/Page.vue [hunk 2]
```diff
@@ -1,7 +1,158 @@
...
+    await useFetch('/api/page_donation',{
+    method:'POST'
+})
+const page_data_cookie=useCookie('cv');
+const page_data= computed (() => JSON.parse(page_data_cookie.value || "{}"))
+const router=useRoute()
+const id= computed(() => {router.params.id});
```
Comment (pariahGH) [inline pages/Page.vue:56]: See ofeks work in EditPage/[id].vue for getting this id

File: pages/Page.vue [hunk 3]
```diff
@@ -1,7 +1,158 @@
...
+	
+}
+
+</script>
 
 <template lang="pug">
-div This is where a page will be rendered. Use route or query parameters to identify which page needs to be loaded.
+doctype html 
+head 
+	link(rel='preconnect' href='https://fonts.googleapis.com/')
```
Comment (pariahGH) [inline pages/Page.vue:71]: Don't need doctype, html, head stuff here - this is provided by Nuxt in a separate location

File: pages/Page.vue [hunk 4]
```diff
@@ -1,7 +1,158 @@
...
+				a.font-poppins.text-center.font-semibold(style="color:#383838; text-shadow: 2px 2px 4px rgba(0, 0, 0, 0.25);" href="#services") Services
+			li.text-xs.text-center.px-auto.py-2(class="w-1/4")
+				a.font-poppins.text-center.font-semibold(style="color:#383838; text-shadow: 2px 2px 4px rgba(0, 0, 0, 0.25);" href="#donations") Donations
+			li.text-xs.text-center.px-auto.py-2(class="w-1/4")
+				a.font-poppins.text-center.font-semibold(style="color:#383838; text-shadow: 2px 2px 4px rgba(0, 0, 0, 0.25);" href="#media") Media
+
+		.div(style="background: #F8F8F8;" id="tab-contents")
+			.div.px-8.py-4(style="color: #6E6E6E; font-weight: 500; font-size: 14px; line-height: 28px; letter-spacing: -0.078px; word-break: break-word;" id="obituary") {{data.obituary}} Obituary 
+			.div.p-4(id="services")
+				if visitation_date
```
Comment (pariahGH) [inline pages/Page.vue:102]: Update template usage - this needs to use Vue's interpolation and [conditional rendering](https://vuejs.org/guide/essentials/conditional.html)

File: pages/Page.vue [hunk 5]
```diff
@@ -1,7 +1,158 @@
...
+					.px-5.py-4.text-gray-dark.font-poppins.text-2xl.text-left.font-bold(style="line-height: 36px; text-shadow: 3px 3px 4px rgba(0, 0, 0, 0.25);") Visitation
+					.px-5.pt-2.pb-8.font-outfit.text-dark-blue(style="font-size: 20px; line-height: 30px;") {{data.visation_description}}
+					.grid.gap-x-3.gap-y-0.grid-cols-5.grid-rows-2
+						img.px-4(src="/images/clock_icon.png")
+						.py-1.font-outfit.text-dark-blue.col-span-4(style="font-size: 20px; line-height: 30px;") {{data.visitation_date + ", " + data.timezone}}
+						img.px-2(src="/images/location_icon.png")
+						.py-2.font-outfit.text-dark-blue.col-span-4.whitespace-normal(style="font-size: 20px; line-height: 30px;" id="visitation_location") visitation location
+				if funeral_date
```
Comment (pariahGH) [inline pages/Page.vue:110]: Update template usage - this needs to use Vue's interpolation and conditional rendering

File: pages/Page.vue [hunk 6]
```diff
@@ -1,7 +1,158 @@
...
+						.progress-bar.overflow-hidden.ml-4.h-5.rounded-full(style="30px; background-color:#b5b5b5;")
+							.progress.rounded-full.text-white.flex.items-center.justify-center(donated-amount=donated_percentage style="background: linear-gradient(90deg, rgba(15,200,0,1) 35%, rgba(203,255,0,1) 100%); box-shadow: 0 3px 3px -5px #1ba710, 0 2px 5px #1ba710; height: 100%; opacity: 0; width:0px;") {{data.donated_percentage + "%"}}
+						.well.well-sm
+							h1.ml-4.pt-9.text-2xl.text-gray-dark(class="sm:text-3xl" style="font-weight: 600; letter-spacing: 0.35px;") Donor Information
+								.bar(style="border-top: 0.5px solid #646464;")
+						br
+						form.form-horizontal(method='post' action=userAction)
+							fieldset
+								.col-md-8.ml-4.pt-1.pr-5(class="sm:mx-4 sm:w-full sm:py-2")
+									input#first_name.rounded-md.outline-0.border-box.w-full.p-2(style="border: 1px solid #c4c4c4;" v-model='first_name' type='text' placeholder='First Name' required)
```
Comment (pariahGH) [inline pages/Page.vue:131]: The v-models of the form need to point at properties on the data object

File: pages/Page.vue [hunk 7]
```diff
@@ -1,7 +1,158 @@
...
+									button#submit.mx-auto.p-3.px-6.pt-2.bg-orange.text-md(style="color: white; font-weight: 700; border-radius: 32px;" name='submit') DONATE NOW
+			.div.p-4(id="media") 
+				.row.gallery.flex.flex-wrap.gap-1.items-center.justify-center(class="basis-1/2 sm:basis-1/4 sm:gap-3 sm:m-8")
+					each image in media ? media : []
+						.div(style='position: relative')
+							img.object-cover.align-middle.rounded-lg(class="w-40 sm:w-64" src = image)
+							
+								 
+
+html
```
Comment (pariahGH) [inline pages/Page.vue:154]: Unnecessary bits

File: pages/Page.vue [hunk 8]
```diff
@@ -1,7 +1,158 @@
...
+								.col-md-8.ml-4.pt-4.pr-5.grid.grid-cols-3(class="sm:mx-4 sm:w-full sm:py-2")
+									span.rounded-l-md.p-3.col-span-2(style="text-shadow: 3px 3px 4px rgba(0, 0, 0, 0.25); border: 1px solid #c4c4c4;") Donation Amount
+									.flex
+										span.bg-gray-light.py-2.px-1.text-lg(style="text-shadow: 3px 3px 4px rgba(0, 0, 0, 0.25); border: 1px solid #c4c4c4; border-right:none;") $
+										input#donation_amount.bg-gray-light.outline-0.rounded-r-md.border-box.w-full.p-2(style="border: 1px solid #c4c4c4; border-left:none;" v-model='donation_amount' type="number" min="0.00" step="0.01" required)
+								.col-md-8.ml-4.pt-6.pr-5.flex.items-center.justify-center
+									button#submit.mx-auto.p-3.px-6.pt-2.bg-orange.text-md(style="color: white; font-weight: 700; border-radius: 32px;" name='submit') DONATE NOW
+			.div.p-4(id="media") 
+				.row.gallery.flex.flex-wrap.gap-1.items-center.justify-center(class="basis-1/2 sm:basis-1/4 sm:gap-3 sm:m-8")
+					each image in media ? media : []
```
Comment (pariahGH) [inline pages/Page.vue:148]: Use  [list rendering](https://vuejs.org/guide/essentials/list.html)

File: pages/Page.vue [hunk 9]
```diff
@@ -1,7 +1,158 @@
...
+doctype html 
+head 
+	link(rel='preconnect' href='https://fonts.googleapis.com/')
+	link(rel='preconnect' href='https://fonts.gstatic.com/' crossorigin='')
+	link(href='https://fonts.googleapis.com/css2?family=Poppins:ital@0;1&display=swap' rel='stylesheet')
+	link(href='https://fonts.googleapis.com/css2?family=Outfit:wght@600&display=swap' rel='stylesheet')
+html
+	head
+		
+		link(rel='stylesheet' href='/stylesheets/main.css')
```
Comment (pariahGH) [inline pages/Page.vue:78]: dont need to import stylesheets, this project uses tailwind

File: pages/Page.vue (hunk 1 above)
Comment (pariahGH) [reply pages/Page.vue:49]: Also note that you will need to implementing loading of the page data based on the route param

File: pages/Page.vue (hunk 3 above)
Comment (namra95) [reply pages/Page.vue:71]: done

File: pages/Page.vue [hunk 10]
```diff
@@ -1,7 +1,158 @@
...
+    obituary: String,
+    deadline: Date,
+    donation_goal: Number,
+    donated_amount: Number,
+    donated_percentage: Number,
+    query: String
+
+}
+const data = ref<Page>( {
+    Cuid : "",
```
[the remark this answers is not in the document]
Comment (namra95) [reply pages/Page.vue:30]: done

File: pages/Page.vue (hunk 5 above)
Comment (namra95) [reply pages/Page.vue:110]: so use v-if instead of if here?

**Pull request merged**